packages/@aws-cdk/aws-s3/lib/rule.ts

```diff
@@ -142,6 +142,38 @@ export interface LifecycleRule {
   readonly objectSizeGreaterThan?: number;
 }
 
+/**
+ * A point in time when an object should transition to a different storage class.
+ */
+export class TransitionTime {
+
+  /**
+   * A moment defined by a specific date.
+   *
+   * The date value must be in ISO 8601 format. The time is always midnight UTC.
```

Comment:
Well the date is not in ISO8601 format -- it is in `Date` format :D 

Honestly, JavaScript `Date`s are very untested and have unpredictable timezone issues. I wonder if this is worth having, or if we should just ask people to put in date ISO8601 date strings.

Reply:
Fair enough. I just copied the docstring from the existing properties.